File: core/src/com/infeez/simple/utils/BoardCommandUtil.java
package com.infeez.simple.utils;

import java.util.HashMap;
import java.util.Map;

public class BoardCommandUtil {

    private static final Map<Character, Integer> xPanel = new HashMap<>();
    private static final Map<Character, Integer> yPanel = new HashMap<>();

    static {
        xPanel.put('a', 0);
        xPanel.put('b', 1);
        xPanel.put('c', 2);
        xPanel.put('d', 3);
        xPanel.put('e', 4);
        xPanel.put('f', 5);
        xPanel.put('g', 6);
        xPanel.put('h', 7);

        yPanel.put('1', 7);
        yPanel.put('2', 6);
        yPanel.put('3', 5);
        yPanel.put('4', 4);
        yPanel.put('5', 3);
        yPanel.put('6', 2);
        yPanel.put('7', 1);
        yPanel.put('8', 0);
    }

    public static BoardArrayPosition parseCommand(String command) {
        if (command.length() != 2) {
            throw new IllegalStateException("The command string length must be 2!");
        }
        BoardArrayPosition boardArrayPosition = new BoardArrayPosition();
        boardArrayPosition.setIndexFirst(xPanel.get(command.charAt(0)));
        boardArrayPosition.setIndexSecond(yPanel.get(command.charAt(1)));
        return boardArrayPosition;
    }

    public static String checkerPositionToCommand(BoardArrayPosition boardArrayPosition) {
        final StringBuilder result = new StringBuilder(2);
        for (Map.Entry<Character, Integer> entry : xPanel.entrySet()) {
            if (entry.getValue() == boardArrayPosition.getIndexFirst()) {
                result.append(entry.getKey());
            }
        }
        for (Map.Entry<Character, Integer> entry : yPanel.entrySet()) {
            if (entry.getValue() == boardArrayPosition.getIndexFirst()) {
                result.append(entry.getKey());
            }
        }

        return result.toString();
    }


}

File: core/src/main/kotlin/com/infeez/simple/utils/BoardCommandUtil.kt
package com.infeez.simple.utils

object BoardCommandUtil {
    private const val FIRST_FILE = 'a'
    private const val LAST_FILE = 'h'
    private const val FIRST_RANK = '1'
    private const val LAST_RANK = '8'

    @JvmStatic
    fun parseCommand(command: String?): BoardArrayPosition {
        require(!command.isNullOrBlank()) {
            "Board command must not be null, empty, or blank."
        }
        require(command.length == 2) {
            "Board command must be exactly 2 characters."
        }

        val file = command[0]
        val rank = command[1]
        require(file in FIRST_FILE..LAST_FILE) {
            "Board command file must be in range a..h."
        }
        require(rank in FIRST_RANK..LAST_RANK) {
            "Board command rank must be in range 1..8."
        }

        return BoardArrayPosition(
            indexFirst = file - FIRST_FILE,
            indexSecond = BoardConfig.BOARD_SIZE - rank.digitToInt(),
        )
    }

    @JvmStatic
    fun checkerPositionToCommand(boardArrayPosition: BoardArrayPosition): String {
        require(boardArrayPosition.indexFirst in 0 until BoardConfig.BOARD_SIZE) {
            "Board file index must be in range 0..7."
        }
        require(boardArrayPosition.indexSecond in 0 until BoardConfig.BOARD_SIZE) {
            "Board rank index must be in range 0..7."
        }

        val file = (FIRST_FILE.code + boardArrayPosition.indexFirst).toChar()
        val rank = BoardConfig.BOARD_SIZE - boardArrayPosition.indexSecond
        return "$file$rank"
    }
}
